matrix/build.gradle

```diff
@@ -18,53 +18,85 @@ dependencies {
   implementation("io.opentelemetry:opentelemetry-extension-annotations:${versions["opentelemetry"]}")
 }
 
-def dockerWorkingDir = new File(project.buildDir, "docker")
-
 def buildProprietaryTestImagesTask = tasks.create("buildProprietaryTestImages") {
   group = "build"
   description = "Builds all Docker images for the test matrix for proprietary app servers"
 }
 
-def targets = [
-   "weblogic" : [
+def proprietaryTargets = [
+    "weblogic" : [
        "12.1.3" : ["developer"],
        "12.2.1.4" : ["developer"],
         "14.1.1.0" : ["developer-8", "developer-11"]
-   ],
+    ],
     "jboss-eap" : [
         "7.1.0" : ["8"],
         "7.3.0" : ["8", "11"]
     ]
 ]
 
-targets.forEach { server, data ->
-  String dockerFileName = "${server}.dockerfile"
-  data.forEach { version, jdks ->
-    jdks.forEach { jdk ->
-      def prepareTask = tasks.register("${server}ImagePrepare-$version-jdk$jdk", Copy) {
-        def warTask = project.tasks.named("war").get()
-        it.dependsOn(warTask)
-        it.into(dockerWorkingDir)
-        it.from("src/${dockerFileName}")
-        it.from("src/main/docker/${server}")
-        it.from(warTask.archiveFile) {
-          rename { _ -> "app.war" }
+createDockerTasks(buildProprietaryTestImagesTask, proprietaryTargets, false)
+
+def buildWindowsTestImagesTask = tasks.create("buildWindowsTestImages") {
+  group = "build"
+  description = "Builds all Windows Docker images for the test matrix"
+}
+
+def windowsTargets = [
+    "tomcat" : [
+        [version: "7.0.107", majorVersion: "7"]: ["8"],
```

Comment:
Why in case of Windows we have separate `version` and `majorVersion`?

Reply:
Anything beyond `version` (used for image name), `classifier` (used to choose which Dockerfile to use) is just an argument directly passed to the `Dockerfile`, so that we can have clean appserver name and version fields without having to leak Dockerfile specific details into those fields.

Reply:
Just to specify, for Tomcat this is necessary because the major version alone is also used in the URLs, so it makes more sense to pass this in as an argument than try to perform string manipulation within a Dockerfile.